adc/onboard_temperature/onboard_temperature.c

```diff
@@ -0,0 +1,48 @@
+#include <stdio.h>
+#include "pico/stdlib.h"
+#include "hardware/adc.h"
+
+/* References for this implementation:
+ * raspberry-pi-pico-c-sdk.pdf, page 68. 
+ * pico-examples/adc/adc_console/adc_console.c */
+const float read_onboard_temperature(const char unit) {
+    
+    /* 12-bit conversion, assume max value == ADC_VREF == 3.3 V */
+    const float conversionFactor = 3.3f / (1 << 12);
+
+    adc_select_input(4);
+    float adc = adc_read() * conversionFactor;
+
+    float tempC = 27.0 - (adc - 0.706) / 0.001721;
+
+    if (unit == 'C') {
+        return tempC;
+    } else if (unit == 'F') {
+        return tempC * 9 / 5 + 32;
+    }
+
+    return -1.0;
+}
+
+int main() {
+    const uint LED_PIN = 25;
+
+    stdio_init_all();
+    gpio_init(LED_PIN);
+    gpio_set_dir(LED_PIN, GPIO_OUT);
+
+    adc_init();
+    adc_set_temp_sensor_enabled(true);
+
+    while (true) {
+        printf("Onboard temperature = %.02f C\n", read_onboard_temperature('C'));
```

Comment:
Perhaps it makes sense to add a `#define TEMPERATURE_UNITS 'C'` near the top of this file, and then this line could become:
```
printf("Onboard temperature = %.02f " TEMPERATURE_UNITS "\n", read_onboard_temperature(TEMPERATURE_UNITS));
```

Although code like this is sometimes easier to debug if split over separate lines like:
```
float temperature = read_onboard_temperature(TEMPERATURE_UNITS);
printf("Onboard temperature = %.02f " TEMPERATURE_UNITS "\n", temperature);
```